src/config-loader/config-loader.ts

```diff
@@ -52,294 +52,267 @@ export interface Config {
+			const authApiTx = sbvrUtils.api.Auth.clone({
+				passthrough: {
+					tx,
+					req: permissions.root,
+				},
+			});
 
-					if (data.users != null) {
-						const permissionsCache: {
-							[index: string]: Bluebird<number>;
-						} = {};
-						_.each(data.users, user => {
-							if (user.permissions == null) {
-								return;
-							}
-							_.each(user.permissions, permissionName => {
-								if (permissionsCache[permissionName] != null) {
-									return;
-								}
-								permissionsCache[permissionName] = authApiTx
-									.get({
+			const { users } = data;
+			if (users != null) {
+				const permissionsCache: {
+					[index: string]: Promise<number>;
+				} = {};
+				_.each(users, user => {
+					if (user.permissions == null) {
+						return;
+					}
+					_.each(user.permissions, permissionName => {
+						if (permissionsCache[permissionName] != null) {
+							return;
+						}
+						permissionsCache[permissionName] = (async () => {
+							try {
+								const result = (await authApiTx.get({
+									resource: 'permission',
+									options: {
+										$select: 'id',
+										$filter: {
+											name: permissionName,
+										},
+									},
+								})) as Array<{ id: number }>;
+								if (result.length === 0) {
+									const { id } = (await authApiTx.post({
 										resource: 'permission',
-										options: {
-											$select: 'id',
-											$filter: {
-												name: permissionName,
-											},
+										body: {
+											name: permissionName,
 										},
-									})
-									.then((result: sbvrUtils.AnyObject[]) => {
-										if (result.length === 0)
-											return authApiTx
-												.post({
-													resource: 'permission',
-													body: {
-														name: permissionName,
-													},
-													options: { returnResource: false },
-												})
-												.then(({ id }: sbvrUtils.AnyObject) => id);
-										else {
-											return result[0].id;
-										}
-									})
-									.tapCatch(e => {
-										e.message = `Could not create or find permission "${permissionName}": ${e.message}`;
-									});
-							});
-						});
+										options: { returnResource: false },
+									})) as { id: number };
+									return id;
+								} else {
+									return result[0].id;
+								}
+							} catch (e) {
+								e.message = `Could not create or find permission "${permissionName}": ${e.message}`;
+								throw e;
+							}
+						})();
+					});
+				});
 
-						return Bluebird.map(data.users, user => {
-							return authApiTx
-								.get({
-									resource: 'user',
+				await Bluebird.map(users, async user => {
+					try {
+						const result = (await authApiTx.get({
+							resource: 'user',
+							options: {
+								$select: 'id',
+								$filter: {
+									username: user.username,
+								},
+							},
+						})) as Array<{ id: number }>;
+						let userID: number = result[0].id;
```

Comment:
:grey_exclamation:  We are accessing `result[0].id` before checking the length of the `result`.

Reply:
Yeah, that's my mistake - tried to eliminate an else